File: ExampleResources/killchat/killchat.js
API.onPlayerDeath.connect(function (player, reason, weapon) {
    var killer = "";
    API.consoleOutput("killed: " + player.Name + " reason: " + reason.Value + " weapon: " + weapon);
    if (reason.IsNull)
    {
    }
    else
    {
        var players = API.getAllPlayers();
        for (var i = 0; i < players.Count; i++) {
            if (players[i].CharacterHandle == reason.Value) {
                killer = players[i].Name;
                break;
            }            
        }        
    }
    
    var players = API.getAllPlayers();
    for (var i = 0; i < players.Count; i++) {
        API.triggerClientEvent(players[i], "addKillToKillchat", player.Name, killer, weapon);
    }    
});

File: ExampleResources/killchat/killchat.cs
using System;
using System.Collections.Generic;
using GTANetworkServer;
using GTANetworkShared;

public class KillchatUtility : Script
{
	public KillchatUtility()
	{
		API.onPlayerDeath += onPlayerDeath;
	}

	public void onPlayerDeath(Client player, NetHandle reason, int weapon)
	{
		var killer = "";
	    API.consoleOutput("killed: " + player.name + " reason: " + reason.Value + " weapon: " + weapon);

	    if (!reason.IsNull)	    
	    {
	        var players = API.getAllPlayers();
	        for (var i = 0; i < players.Count; i++)
	        {
	            if (players[i].handle == reason) {
	                killer = players[i].name;
	                break;
	            }            
	        }        
	    }
	    
	    API.triggerClientEventForAll("addKillToKillchat", player.name, killer, weapon);
	}	    
}
